data/datatest/impls.go

```diff
@@ -159,13 +159,9 @@ func (i ledgerImpl) EnsureDigest(cert agreement.Certificate, quit chan struct{},
 		return
 	}
 
-	select {
-	case <-quit:
-		return
-	case <-i.Wait(r):
-		if !consistencyCheck() {
-			err := fmt.Errorf("Wait channel fired without matching block in round %v", r)
-			panic(err)
-		}
+	<-i.Wait(r)
```

Comment:
The actual EnsureDigest is nonblocking now, should this function also be made nonblocking?